how does one load a module consisting of a .js file and .d.ts file?

It seems impossible to do this. I currently have a config.js containing
```json
  packages: {
    "app": {
      "main": "app",
      "defaultExtension": "tsx",
      "meta": {
        "*.tsx": {
          "loader": "ts"
        }
      }
    },
    "app/js": {
    }
}
```
where `app/js` contains `protos.js` and `protos.d.ts`. This configuration results in the application correctly fetching both files (from the network) but then the typescript loader complains that:
```
TypeScript http://localhost:8080/next/app/redux/nodes.ts:14:25
TypeScript Cannot find module '../js/protos'. (TS2307)
```

For context, `js/protos` is being imported from `app/redux/nodes.ts` using `import * as protos from "../js/protos";`.

How can I get this working?

--- a/lib/resolver.js
+++ b/lib/resolver.js
@@ -76,7 +76,7 @@ System.register(['typescript', './logger', './utils'], function(exports_1, conte
                         importName = importName + ".d.ts";
                     return this._resolve(importName, sourceName)
                         .then(function (address) {
-                        if (utils_1.isAmbientImport(importName) && utils_1.isJavaScript(address)) {
+                        if (!utils_1.isAbsolute(importName) && utils_1.isJavaScript(address)) {
                             return _this.lookupTyping(importName, sourceName, address)
                                 .then(function (typingAddress) {
                                 return typingAddress ? typingAddress : address;

--- a/src/resolver.ts
+++ b/src/resolver.ts
@@ -42,7 +42,7 @@ export class Resolver {
                const deps = Object.keys(mappings)
                   .map((key) => mappings[key])
                   .filter((res) => isTypescript(res)) // ignore e.g. js, css files
-					
+
                /* add the fixed declaration files */
                const refs = this._declarationFiles.filter(decl => {
                   return (decl != sourceName) && (deps.indexOf(decl) < 0);
@@ -65,7 +65,7 @@ export class Resolver {
    public registerDeclarationFile(sourceName: string) {
       this._declarationFiles.push(sourceName);
    }
-	
+
 	/*
 		process the source to get its dependencies and resolve and register them
 		returns a promise to a map of import/reference name  -> resolved file

--- a/src/resolver.ts
+++ b/src/resolver.ts
@@ -5,7 +5,7 @@ import {CompilerHost} from './compiler-host';
 import {
    isTypescript, isTypescriptDeclaration,
    isJavaScript, isRelative,
-   isAmbient, isAmbientImport,
+   isAmbient, isAbsolute,
    jsToDts
 } from './utils';
 
@@ -105,7 +105,7 @@ export class Resolver {
 
       return this._resolve(importName, sourceName)
          .then(address => {
-            if (isAmbientImport(importName) && isJavaScript(address)) {
+            if (!isAbsolute(importName) && isJavaScript(address)) {
                return this.lookupTyping(importName, sourceName, address)
                   .then(typingAddress => {
                      return typingAddress ? typingAddress : address;

